storage.go

```diff
@@ -243,13 +243,25 @@ func (s *storage) InsertRows(rows []Row) error {
 
 	insert := func() error {
 		defer func() { <-s.workersLimitCh }()
-		p := s.getPartition()
-		outdatedRows, err := p.insertRows(rows)
-		if err != nil {
-			return fmt.Errorf("failed to insert rows: %w", err)
+		s.ensureActiveHead()
+		iterator := s.partitionList.newIterator()
+		rowsToInsert := rows
+		// Starting at the head partition, try to insert rows, and loop to insert outdated rows
+		// into older partitions. Any rows more than `defaultWritablePartitionsNum` partitions out
+		// of date are dropped.
+		for i := 0; i < defaultWritablePartitionsNum; i++ {
+			if len(rowsToInsert) == 0 {
+				return nil
+			}
+			if !iterator.next() {
+				return nil
+			}
```

Comment:
nit: Currently it's fine, but `break` makes more sense than the early return. To avoid future confusion, let's use it.
```suggestion
			if len(rowsToInsert) == 0 {
				break
			}
			if !iterator.next() {
				break
			}
```